src/main/java/com/puppycrawl/tools/checkstyle/DefaultLogger.java

```diff
@@ -211,4 +214,94 @@ private void closeStreams() {
         }
     }
 
+    /**
+     * Represents a message that can be localised. The translations come from
+     * message.properties files. The underlying implementation uses
+     * java.text.MessageFormat.
+     *
+     * @noinspection SerializableHasSerializationMethods
+     */
+    /* package */ static class LocalizedMessage {
```

Comment:
what force you to make it package visibility ?
it is better to be private or public. All other visibility is problem of design.

Reply:
> what force you to make it package visibility ?

We cannot make it private since, we need to directly use the class for testing of intricate cases which otherwise is not possible. I neither wanted to expose the class as public as it is no longer API and no other classes use it. 

> it is better to be private or public.

I would like to go with `public` in that case.

Reply:
> We cannot make it private since, we need to directly use the class for testing of intricate cases which otherwise is not possible.

it is exactly good point to discuss, as all should be tested through public methods, only and only,  in very rare cases when it is not possible we use reflection to get access to private method as clear signal that edge case.
But we never make meethod more accessible to ease testing. It make design bad and it make test strategy to be useless from functional perspective.

Reply:
Done